_posts/2025-05-09-query-insights-live-queries.md

```diff
@@ -0,0 +1,141 @@
+
+### Basic request
+
+To get a list of currently running queries sorted by latency, send the following request:
+
+```json
+GET /_insights/live_queries
+```
+
+### Customizing your view using query parameters
+
+You can tailor the output using several optional query parameters.
+
+| Parameter | Data type | Description                                                                                                |
+| :-------- | :-------- | :--------------------------------------------------------------------------------------------------------- |
+| `verbose` | Boolean   | Whether to include detailed query information (like the query source) in the output. Default is `true`.    |
+| `nodeId`  | String    | A comma-separated list of node IDs to filter results. If omitted, queries from all nodes are returned.     |
+| `sort`    | String    | The metric to sort results by. Valid values: `latency`, `cpu`, or `memory`. Default is `latency`.          |
+| `size`    | Integer   | The number of query records to return. Default is `100`.                                                     |
+
+### Example: Finding CPU-intensive live queries
+
+To find the top 5 queries currently consuming the most CPU, you can provide the following query parameters:
+
+```json
+GET /_insights/live_queries?verbose=false&sort=cpu&size=5
+```
+{% include copy-curl.html %}
+
+### Understanding the response
+
+Here’s an example of what the response might look like (showing one query for brevity, taken from the [documentation example](https://docs.opensearch.org/docs/latest/observing-your-data/query-insights/live-queries/#example-response)):
+
+```json
+{
+  "live_queries" : [
+    {
+      "timestamp" : 1745359226777,
+      "id" : "troGHNGUShqDj3wK_K5ZIw:512",
+      "description" : "indices[my-index-*], search_type[QUERY_THEN_FETCH], source[{\"size\":20,\"query\":{\"term\":{\"user.id\":{\"value\":\"userId\",\"boost\":1.0}}}}]",
+      "node_id" : "troGHNGUShqDj3wK_K5ZIw",
+      "measurements" : {
+        "latency" : {
+          "number" : 13959364458,
+          "count" : 1,
+          "aggregationType" : "NONE"
+        },
+        "memory" : {
+          "number" : 3104,
+          "count" : 1,
+          "aggregationType" : "NONE"
+        },
+        "cpu" : {
+          "number" : 405000,
+          "count" : 1,
+          "aggregationType" : "NONE"
+        }
+      }
+    }
+    // ... other live queries
+  ]
+}
+```
+
+This response provides the following information:
+
+*   The query started at timestamp `1745359226777`.
+*   It's running on node `troGHNGUShqDj3wK_K5ZIw`.
+*   So far, it has been running for over 13.9 seconds (`latency.number` in nanoseconds).
+*   It has consumed `405000` nanoseconds of CPU time and `3104` bytes of memory.
+*   The `description` (provided because `verbose=true` in the request) shows the query targets `my-index-*` and includes the actual query structure.
```

Comment:
```suggestion
*   The `description` (provided because `verbose=true` in the request) shows that the query targets `my-index-*` and includes the actual query structure.
```